includes/embeds/class-amp-dailymotion-embed.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+require_once( AMP__DIR__ . '/includes/embeds/class-amp-base-embed-handler.php' );
+
+// Much of this class is borrowed from Jetpack embeds
+class AMP_DailyMotion_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	const URL_PATTERN = '#https?:\/\/(www\.)?dailymotion\.com\/video\/.*#i';
+	const RATIO = 0.5625;
+
+	protected $DEFAULT_WIDTH = 600;
+	protected $DEFAULT_HEIGHT = 338;
+
+	private static $script_slug = 'amp-dailymotion';
+	private static $script_src = 'https://cdn.ampproject.org/v0/amp-dailymotion-0.1.js';
+
+	function __construct( $args = array() ) {
+		parent::__construct( $args );
+
+		if ( isset( $this->args['content_max_width'] ) ) {
+			$max_width = $this->args['content_max_width'];
+			$this->args['width'] = $max_width;
+			$this->args['height'] = round( $max_width * self::RATIO );
+		}
+	}
+
+	function register_embed() {
+		wp_embed_register_handler( 'amp-dailymotion', self::URL_PATTERN, array( $this, 'oembed' ), -1 );
+		add_shortcode( 'dailymotion', array( $this, 'shortcode' ) );
+	}
+
+	public function unregister_embed() {
+		wp_embed_unregister_handler( 'amp-dailymotion', -1 );
+		remove_shortcode( 'dailymotion' );
+	}
+
+	public function get_scripts() {
+		if ( ! $this->did_convert_elements ) {
+			return array();
+		}
+
+		return array( self::$script_slug => self::$script_src );
+	}
+
+	public function shortcode( $attr ) {
+		$url = false;
+
+		if ( isset( $attr[0] ) ) {
+			$url = ltrim( $attr[0] , '=' );
+		} elseif ( function_exists ( 'shortcode_new_to_old_params' ) ) {
+			$url = shortcode_new_to_old_params( $attr );
+		}
+
+		if ( empty( $url ) ) {
+			return '';
+		}
+
+		$video_id = $this->get_video_id_from_url( $url );
+
+		return $this->render( array(
+			'url' => $url,
+			'video_id' => $video_id,
+		) );
```

Comment:
It looks like the shortcode for Dailymotion expects the `id` not the URL, so we may need to modify some of the handling here.

The `oembed` method should call `get_video_id_from_url` to extract the `video_id` and then call `render` directly.

The `shortcode` method can mostly stay as-is, expect we can change references from `$url` => `$video_id` and remove the call to `get_video_id_from_url`. We also need to add a branch to the first `if` block to look for `$attr['id']` and use that if set.

The tests should be updated accordingly as well.